Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -348,52 +348,112 @@ Extbase controller.
 
    $includeCategories = (bool) ($this->settings['includeCategories'] ?? false);
 
-
 .. important::
 
    If you wish to access a setting from your controller via
    :php:`$this->settings`, the name of the setting must be prefixed with `settings.`,
    so literally `settings` directly followed by a dot (`.`).
 
-
-.. index:: pair: Flexforms; PHP
+..  index::
+    FlexForms; Read in PHP
+    FlexForms; FlexFormService
 .. _read-flexforms-php:
 
-How to read and write flexforms from PHP
-----------------------------------------
+Read FlexForms values in PHP
+----------------------------
 
-Some situation make it necessary to access Flexforms via PHP. The following APIs
-are available to work with Flexforms from within PHP:
+You can use the :php:`FlexFormService` to read the content of a FlexForm field:
 
-In order to convert a Flexform to a PHP array, the :php:`xml2array` method can
-be used:
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/NonExtbaseController
 
-.. code-block:: php
+    use TYPO3\CMS\Core\Service\FlexFormService;
+    use TYPO3\CMS\Core\Utility\GeneralUtility;
 
-   $flexFormArray = \TYPO3\CMS\Core\Utility\GeneralUtility::xml2array($flexFormString);
+    // Inject FlexFormService
+    public function __construct(
+        protected readonly FlexFormService $flexFormService,
+    ) {
+    }
 
+    private function loadFlexForm(): void
+    {
+        $this->flexFormData = $this->flexFormService
+            ->convertFlexFormContentToArray($this->cObj->data['pi_flexform']);
+    }
 
-In order to convert an PHP array into an Flexform, the :php`flexArray2Xml`
-method can be used:
+The resulting array would look this:
+
+..  code-block:: php
+    :caption: Example output
+
+    $output = $this->flexFormService
+            ->convertFlexFormContentToArray($this->cObj->data['pi_flexform']);
+
+    $looksLike = [
+        'settings' => [
+            'singlePid' => 25,
+            'listPid' => 380,
+        ],
+    ];
+
+While the result of :php:`GeneralUtility::xml2array()` might look like:
```

Comment:
I don't get the way from the FlexFormService to xml2array. Maybe explain, that the FlexForm settings can be retrieved in two ways: 1. with FlexFormService, 2. with xml2array(). Aaaand: do we need the xml2array explanation really? This has less value then the FlexFormService for a developer when retrieving the values of a flexform.